### ISSUE TITLE
net.LookupIP fails

### ISSUE THREAD

**Issue by rogpeppe** [rogpeppe/godef#41]
Trying to find the definition of LookupIP in the following code fails:

```
package main
import "net"

var x = net.LookupIP

func main() {
}
```

Looking at the debug output, it's probably something to do with build tags:

```
2016/06/20 10:36:56 exprType tuple:false pkg: *ast.SelectorExpr net.LookupIP [
2016/06/20 10:36:56 exprType tuple:false pkg: *ast.Ident net [
2016/06/20 10:36:56 exprType tuple:false pkg: *ast.ImportSpec "net" [
2016/06/20 10:36:56 ] -> 0x0, Type{package "" *ast.ImportSpec "net"}
2016/06/20 10:36:56 ] -> 0xc42010d950, Type{package "" *ast.ImportSpec "net"}
2016/06/20 10:36:56 member Type{package "" *ast.ImportSpec "net"} 'LookupIP' {
2016/06/20 10:36:56     /home/rog/go/src/net/interface_bsdvar.go:12:2: cannot find identifier for package "golang.org/x/net/route": no buildable Go source files in /home/rog/go/src/vendor/golang.org/x/net/route
2016/06/20 10:36:56 } -> <nil>
2016/06/20 10:36:56 ] -> 0x0, Type{bad "" <nil> }
2016/06/20 10:36:56 exprType tuple:false pkg: *ast.SelectorExpr net.LookupIP [
2016/06/20 10:36:56 exprType tuple:false pkg: *ast.Ident net [
2016/06/20 10:36:56 exprType tuple:false pkg: *ast.ImportSpec "net" [
2016/06/20 10:36:56 ] -> 0x0, Type{package "" *ast.ImportSpec "net"}
2016/06/20 10:36:56 ] -> 0xc42010d950, Type{package "" *ast.ImportSpec "net"}
2016/06/20 10:36:56 member Type{package "" *ast.ImportSpec "net"} 'LookupIP' {
2016/06/20 10:36:56     /home/rog/go/src/net/interface_bsdvar.go:12:2: cannot find identifier for package "golang.org/x/net/route": no buildable Go source files in /home/rog/go/src/vendor/golang.org/x/net/route
2016/06/20 10:36:56 } -> <nil>
2016/06/20 10:36:56 ] -> 0x0, Type{bad "" <nil> }
```

**Comment by zh-lab** [comment]
```
godef -debug -f /usr/local/Cellar/go/1.6.2/libexec/src/log/syslog/syslog.go -o 3403
2016/07/05 13:57:07 exprType tuple:false pkg: *ast.SelectorExpr net.Dial [
2016/07/05 13:57:07 exprType tuple:false pkg: *ast.Ident net [
2016/07/05 13:57:07 exprType tuple:false pkg: *ast.ImportSpec "net" [
2016/07/05 13:57:07 ] -> 0x0, Type{package "" *ast.ImportSpec "net"}
2016/07/05 13:57:07 ] -> 0xc8201de690, Type{package "" *ast.ImportSpec "net"}
2016/07/05 13:57:07 member Type{package "" *ast.ImportSpec "net"} 'Dial' {
2016/07/05 13:57:07 	/usr/local/Cellar/go/1.6.2/libexec/src/net/cgo_android.go:10:8: cannot find identifier for package "C": cannot find package "C" in any of:
	/usr/local/Cellar/go/1.6.2/libexec/src/vendor/C (vendor tree)
	/usr/local/Cellar/go/1.6.2/libexec/src/C (from $GOROOT)
	/Users/Zhiheng/Workspace/go/src/C (from $GOPATH)
2016/07/05 13:57:07 } -> <nil>
2016/07/05 13:57:07 ] -> 0x0, Type{bad "" <nil> }
2016/07/05 13:57:07 exprType tuple:false pkg: *ast.SelectorExpr net.Dial [
2016/07/05 13:57:07 exprType tuple:false pkg: *ast.Ident net [
2016/07/05 13:57:07 exprType tuple:false pkg: *ast.ImportSpec "net" [
2016/07/05 13:57:07 ] -> 0x0, Type{package "" *ast.ImportSpec "net"}
2016/07/05 13:57:07 ] -> 0xc8201de690, Type{package "" *ast.ImportSpec "net"}
2016/07/05 13:57:07 member Type{package "" *ast.ImportSpec "net"} 'Dial' {
2016/07/05 13:57:07 	/usr/local/Cellar/go/1.6.2/libexec/src/net/cgo_android.go:10:8: cannot find identifier for package "C": cannot find package "C" in any of:
	/usr/local/Cellar/go/1.6.2/libexec/src/vendor/C (vendor tree)
	/usr/local/Cellar/go/1.6.2/libexec/src/C (from $GOROOT)
	/Users/Zhiheng/Workspace/go/src/C (from $GOPATH)
2016/07/05 13:57:07 } -> <nil>
2016/07/05 13:57:07 ] -> 0x0, Type{bad "" <nil> }
godef: no declaration found for net.Dial
```

Is this one related here?

**Comment by hotpxl** [comment]
I'm having a related issue on Arch Linux with go1.6.3 linux/amd64

When I try `godef --debug -f abc.go net.DialTimeout`, it returns
```
2016/07/27 20:45:52 exprType tuple:false pkg: *ast.SelectorExpr net.DialTimeout [
2016/07/27 20:45:52 exprType tuple:false pkg: *ast.Ident net [
2016/07/27 20:45:52 exprType tuple:false pkg: *ast.ImportSpec "net" [
2016/07/27 20:45:52 ] -> 0x0, Type{package "" *ast.ImportSpec "net"}
2016/07/27 20:45:52 ] -> 0xc820143a40, Type{package "" *ast.ImportSpec "net"}
2016/07/27 20:45:52 member Type{package "" *ast.ImportSpec "net"} 'DialTimeout' {
2016/07/27 20:45:52     /usr/lib/go/src/net/conf_netcgo.go:14:8: cannot find identifier for package "C": cannot find package "C" in any of:
        /usr/lib/go/src/vendor/C (vendor tree)
        /usr/lib/go/src/C (from $GOROOT)
        /home/hotpxl/go/src/C (from $GOPATH)
2016/07/27 20:45:52 } -> <nil>
2016/07/27 20:45:52 ] -> 0x0, Type{bad "" <nil> }
2016/07/27 20:45:52 exprType tuple:false pkg: *ast.SelectorExpr net.DialTimeout [
2016/07/27 20:45:52 exprType tuple:false pkg: *ast.Ident net [
2016/07/27 20:45:52 exprType tuple:false pkg: *ast.ImportSpec "net" [
2016/07/27 20:45:52 ] -> 0x0, Type{package "" *ast.ImportSpec "net"}
2016/07/27 20:45:52 ] -> 0xc820143a40, Type{package "" *ast.ImportSpec "net"}
2016/07/27 20:45:52 member Type{package "" *ast.ImportSpec "net"} 'DialTimeout' {
2016/07/27 20:45:52     /usr/lib/go/src/net/conf_netcgo.go:14:8: cannot find identifier for package "C": cannot find package "C" in any of:
        /usr/lib/go/src/vendor/C (vendor tree)
        /usr/lib/go/src/C (from $GOROOT)
        /home/hotpxl/go/src/C (from $GOPATH)
2016/07/27 20:45:52 } -> <nil>
2016/07/27 20:45:52 ] -> 0x0, Type{bad "" <nil> }
godef: no declaration found for net.DialTimeout
```

Looks like it's having problems importing cgo

**Comment by muirrn** [comment]
#44 fixed this issue for me.

**Comment by sky126** [comment]
github.com\rogpeppe\godef\go\parser\parser.go

// p.error(path.Pos(), fmt.Sprintf("cannot find identifier for package %q: %v", litToString(path), err))

fixed this

**Comment by rongyi** [comment]
master still has this problem

**Comment by bboreham** [comment]
It seems to me that there is a mismatch between two implementations of "exclude files that are not for the current architecture".

One is in `godef/go/types/goodarch.go:goodOSArch()`, which decides purely on the file name.
The other is in `go/build/build.go:shouldBuild()` via `Context.match()` which looks at `// +build` lines.

So for the example at the top the former decides `/home/rog/go/src/net/interface_bsdvar.go` is ok, and the latter decides there are no files to parse in `/home/rog/go/src/vendor/golang.org/x/net/route`, because they are all excluded by `+build` directives.

Given that `build.Default.Import()` returns a set of Go files and ignored Go files, could we not use them instead of the `isGoFile` logic?

**Comment by rogpeppe** [comment]
Fixed by #62
